store/tikv/snapshot.go

```diff
@@ -557,3 +577,66 @@ func prettyWriteKey(buf *bytes.Buffer, key []byte) {
 		logutil.BgLogger().Error("error", zap.Error(err4))
 	}
 }
+
+func (s *tikvSnapshot) recordBackoffInfo(bo *Backoffer) {
+	if s.stats == nil || bo.totalSleep == 0 {
+		return
+	}
+	s.mu.Lock()
```

Comment:
should we move s.mu.Lock() before L582?

Reply:
No need for this. `s.stats == nil` means doesn't collect the runtime stats.  `bo.totalSleep == 0` no need to do anything.